ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,72 @@
+from collections import OrderedDict
+from collections.abc import Mapping
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, ndcube):
+        """
+        Init method.
+        """
+        super().__init__()
+        self._ndcube = ndcube
+
+        self._internal_coords = OrderedDict()
+
+    @property
+    def _all_coords(self):
+        """
+        Establish _all_coords as a property returning some _internal_coords.
+        """
+        return self._internal_coords
+
+    def add(self, name, physical_type, coords):
+        """
+        Add a new coordinate to the collection.
+        """
+        self._internal_coords[name] = (physical_type, coords)
+
+    def remove(self, name):
+        """
+        Remove a coordinate from the collection
+        """
+        del self._internal_coords[name]
```

Comment:
What's the behaviour here if there is no coord called `'name'`?  Does it error or does it just not remove anything?

Reply:
Probably going to return `None` by default, but let me double check and add a test for `NoneType` accordingly if that's the case.

Reply:
I think having a `'name'` would be like a basic requirement in hindsight...

Reply:
Otherwise it wouldn't make sense like a person without a name... 😅

Reply:
**Reality check**: But does it happen in reality though?

Reply:
No.  I mean if you do
```python
>>> gc.keys()  # Only one coordinate called 'Kris'
('Kris',)
>>> gc.remove{'Dan')
???
```
what happens then.

Reply:
Let me add a test to ascertain now... (P.S. We don't want Dan removed)

Reply:
We will then get a `KeyError` thrown.

Reply:
Added an `xfail` (expected to fail) test for this special edge case.

Reply:
@Cadair suggested an improvement over the `xfail` approach.